### ISSUE TITLE
SNSTopic Converting circular structure to JSON

### ISSUE THREAD

**Issue by jamiedavenport** [sst/ion#918]
When using `SNSTopic` I'm getting the following error:

```
TypeError: Converting circular structure to JSON
|      --> starting at object with constructor '_SnsTopic'
|      |     property 'topic' -> object with constructor 'Topic'
|      --- property '__parentResource' closes the circle
|      at JSON.stringify (<anonymous>)
```

```sh
yarn sst version
sst 3.0.68
```

```ts
// sst.config.ts
export default $config({
  app(input) {
    return {
      // [...]
    };
  },
  async run() {
  // [...]

    const topic = new sst.aws.SnsTopic('Topic');

  // [...]
  }
});

**Comment by jamiedavenport** [comment]
Update: this error only happens when the sns topic is linked to another resource

**Comment by jamiedavenport** [comment]
Also the same issue with `Queue`. However I have a potential fix of wrapping the SnsTopic in an output like so:

```ts
// .sst/platform/src/components/aws/sns-topic.ts

private topic: Output<sns.Topic>;

  constructor(
    name: string,
    args: SnsTopicArgs = {},
    opts: ComponentResourceOptions = {},
  ) {
    super(__pulumiType, name, args, opts);

    const parent = this;
    const fifo = normalizeFifo();

    const topic = createTopic();

    this.constructorName = name;
    this.constructorOpts = opts;
    this.topic = topic.arn.apply(() => topic);

    function normalizeFifo() {
      return output(args.fifo).apply((v) => v ?? false);
    }

    function createTopic() {
      return new sns.Topic(
        ...transform(
          args.transform?.topic,
          `${name}Topic`,
          {
            fifoTopic: fifo,
          },
          { parent },
        ),
      );
    }
  }
```

Unsure if this is the best approach. Happy to open it as a PR if it makes sense

**Comment by jayair** [comment]
@jamiedavenport can you add the snippet that causes the error? Including the link as well.

**Comment by jamiedavenport** [comment]
```ts
/// <reference path="./.sst/platform/config.d.ts" />

/* eslint no-undef: 0 */

export default $config({
  app(input) {
    return {
      // ...
    };
  },
  async run() {
    const topic = new sst.aws.SnsTopic('Topic');

    const apiGateway = new sst.aws.ApiGatewayV1(
      'ApiGateway',
      { endpoint: { type: 'regional' } }
    )

    apiGateway.route(
      'POST /',
      {
        handler: '...',
        link: [
          topic
        ],
        environment: {
          // ...
        }
      }
    )

    return {}
  }
});
```

**Comment by JohnG45** [comment]
Same problem exists with `Bus`. Please let me know if you would like me to raise a separate issue. It's a blocker for us migrating to ion currently.

**Comment by JohnG45** [comment]
I've created a PR that I believe fixes this.

https://github.com/sst/ion/pull/982

**Comment by fwang** [comment]
Fixed in v3.0.99
